src/private/privateAPIs.ts

```diff
@@ -111,15 +111,67 @@ export function uploadCustomApp(manifestBlob: Blob, onComplete?: (status: boolea
  * Sends a custom action message to Teams.
  * @param actionName Specifies name of the custom action to be sent
  * @param args Specifies additional arguments passed to the action
+ * @param callback Optionally specify a callback to receive response parameters from the parent
  * @returns id of sent message
  */
 export function sendCustomMessage(
   actionName: string,
   // tslint:disable-next-line:no-any
   args?: any[],
+  // tslint:disable-next-line:no-any
+  callback?: (...args: any[]) => void,
 ): number {
   ensureInitialized();
-  return sendMessageRequest(GlobalVars.parentWindow, actionName, args);
+
+  const messageId = sendMessageRequest(GlobalVars.parentWindow, actionName, args);
+  if (typeof callback === 'function') {
+    GlobalVars.callbacks[messageId] = (...args: any[]): void => {
+      callback.apply(null, args);
+    };
+  }
+  return messageId;
+}
+
+/**
+ * @private
+ * Internal use only
+ * Sends a custom action message to a child window or iframe.
+ * @param actionName Specifies name of the custom action to be sent
+ * @param args Specifies additional arguments passed to the action
+ * @returns id of sent message
+ */
+export function sendCustomMessageToChild(
```

Comment:
Once you rename `sendCustomMessageRequest` to `sendCustomEvent` you can also just merge this method with it. We can always assume that messages/requests flow up and events flow down; therefore, `sendCustomEvent` can always send the message down to the child.